src/plugins/discover/public/application/components/table/table.tsx

```diff
@@ -90,7 +93,11 @@ export function DocViewTable({
       <tbody>
         {Object.keys(flattened)
           .filter((field) => {
-            return !fieldsWithParents.includes(field);
+            const parent = childParentFieldsMap[field];
+            return (
+              !fieldsWithParents.includes(field) ||
+              (parent && !Object.keys(flattened).includes(parent))
```

Comment:
:art: A bit more performant, since we don't need to create the array (of keys), and have a faster lookup than the O(1) we have in the array:

```suggestion
              (parent && flattened.hasOwnProperty(parent))
```